offline_cam/plugin.cpp

```diff
@@ -38,118 +33,51 @@ class offline_cam : public threadloop {
 
     virtual void _p_one_iteration() override {
         duration time_since_start = _m_rtc->now().time_since_epoch();
-        duration begin            = time_since_start;
-        ullong   lookup_time      = std::chrono::nanoseconds{time_since_start}.count() + dataset_first_time;
+        // duration begin            = time_since_start;
+        ullong lookup_time = std::chrono::nanoseconds{time_since_start}.count() + dataset_first_time;
         std::map<ullong, sensor_types>::const_iterator nearest_row;
 
-#ifndef INFINITE_QUEUE
+        // "std::map::upper_bound" returns an iterator to the first pair whose key is GREATER than the argument.
         auto after_row = _m_sensor_data.upper_bound(lookup_time);
```

Comment:
these after_row, next_row, nearest_row names kinda weird?

Reply:
`nearest_row` is the sample that's gonna be published (nearest to the `lookup_time`), I changed `after_row` to `after_nearest_row` to better reflect that it's the one after the `nearest_row`. Deleted next_row.